File: DSA/Animesh_jha_590016813/day_78_question_1.py
class Node:
    def __init__(self, key):
        self.key = key
        self.left = None
        self.right = None

def search(root, key):
    if root is None:
        return False
    
    if root.key == key:
        return True
    elif key < root.key:
        return search(root.left, key)
    else:
        return search(root.right, key)

root = Node(5)
root.left = Node(3)
root.right = Node(7)
root.left.left = Node(2)
root.left.right = Node(4)
root.right.left = Node(6)
root.right.right = Node(8)

print("Yes" if search(root, 4) else "No")   
print("Yes" if search(root, 10) else "No")  

File: DSA/Ayush_Mishra_590014844/Day_78_Question_1.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node *left, *right;
};

struct Node* createNode(int val) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = val;
    newNode->left = newNode->right = NULL;
    return newNode;
}

struct Node* insert(struct Node* root, int val) {
    if (!root) return createNode(val);
    if (val < root->data) root->left = insert(root->left, val);
    else root->right = insert(root->right, val);
    return root;
}

int search(struct Node* root, int key) {
    while (root) {
        if (key == root->data) return 1;
        root = (key < root->data) ? root->left : root->right;
    }
    return 0;
}

int main() {
    int n, val, key;
    struct Node* root = NULL;
    printf("Enter number of nodes: ");
    scanf("%d", &n);
    printf("Enter %d node values:\n", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &val);
        root = insert(root, val);
    }
    printf("Enter key to search: ");
    scanf("%d", &key);
    if (search(root, key)) printf("Yes\n");
    else printf("No\n");
    return 0;
}
